filesystem/filesystem.go

```diff
@@ -102,6 +103,34 @@ func RenameFile(oldPath, newPath string) error {
 	return os.Rename(oldPath, newPath)
 }
 
+// AtomicWriteFile takes a path, filename, a file and the mode that should be
+// associated to the file and writes it atomically to the disk
+func AtomicWriteFile(folder, filename string, file []byte, mode uint32) error {
```

Comment:
why don't we replace the WriteFile implementation to be atomic?

why keep both?